From 3ba122ebb3462395d26d4f8f55dab510a49f5b5e Mon Sep 17 00:00:00 2001
From: Shayne Chen <shayne.chen@mediatek.com>
Date: Wed, 20 Mar 2024 22:56:44 +0800
Subject: [PATCH 115/199] mtk: mt76: mt7996: tmp disable VOW

FW will return failed when legacy 5G station connects after legacy 2G
station, need to check.

Signed-off-by: Shayne Chen <shayne.chen@mediatek.com>
---
 mt7996/mcu.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index 598f32ba..340b6637 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -2369,6 +2369,7 @@ int mt7996_mcu_add_rate_ctrl(struct mt7996_dev *dev,
 	return mt7996_mcu_add_rate_ctrl_fixed(dev, conf, mconf, link_sta, mlink);
 }
 
+#if 0
 static int
 mt7996_mcu_sta_init_vow(struct mt7996_bss_conf *mconf,
 			struct mt7996_link_sta *mlink)
@@ -2402,6 +2403,7 @@ mt7996_mcu_sta_init_vow(struct mt7996_bss_conf *mconf,
 
 	return mt7996_mcu_set_vow_drr_ctrl(phy, mconf, mlink, VOW_DRR_CTRL_STA_ALL);
 }
+#endif
 
 int mt7996_mcu_add_sta(struct mt7996_dev *dev, struct ieee80211_bss_conf *conf,
 		       struct mt7996_bss_conf *mconf,
@@ -2410,7 +2412,7 @@ int mt7996_mcu_add_sta(struct mt7996_dev *dev, struct ieee80211_bss_conf *conf,
 {
 	struct ieee80211_vif *vif = conf->vif;
 	struct sk_buff *skb;
-	int ret;
+	// int ret;
 
 	skb = __mt76_connac_mcu_alloc_sta_req(&dev->mt76, &mconf->mt76,
 					      &mlink->wcid,
@@ -2456,11 +2458,13 @@ int mt7996_mcu_add_sta(struct mt7996_dev *dev, struct ieee80211_bss_conf *conf,
 		mt7996_mcu_sta_bfee_tlv(dev, skb, conf, mconf, link_sta);
 	}
 
+#if 0
 	ret = mt7996_mcu_sta_init_vow(mconf, mlink);
 	if (ret) {
 		dev_kfree_skb(skb);
 		return ret;
 	}
+#endif
 out:
 	return mt76_mcu_skb_send_msg(&dev->mt76, skb,
 				     MCU_WMWA_UNI_CMD(STA_REC_UPDATE), true);
-- 
2.18.0

